app/controllers/api/v3/drafts_controller.rb

```diff
@@ -121,9 +121,13 @@ def update_narrative
         dmp.narrative.purge if (create_params[:narrative].present? || create_params[:remove_narrative].present?) &&
                                dmp.narrative.attached?
 
+        # Remove the narrative from the metadata if it was purged
+        dmp.metadata.fetch('dmp', {})['draft_data']['narrative'] = {} if (create_params[:remove_narrative].present?) 
```

Comment:
you should probably add a check here to make sure 'draft_data' is present. `if dmp.metadata.fetch('dmp', {})['draft_data'].present? && create_params[:remove_narrative].present?`

Reply:
Thanks Brian. I made the update.